### PR TITLE
Add getNormalsData to AbstractMesh and refactor common part with getP…

### PR DESCRIPTION
**Pull request by carolhmj**
…ositionData

Related forum issue: https://forum.babylonjs.com/t/how-to-get-vertex-normals-of-mesh-with-skeleton-morph/29959/12

Playground: playground.babylonjs.com/#SYQW69#1061

### CODE REVIEW

**Review by sebavan — COMMENTED** [review]
Just wondering if a flag in the parameter of the other function would be enough  to only change the transform function ?

**Review by RaananW — COMMENTED** [review]
It seems like this is mostly the same code as getPositionData, but for a different data type. do you think it'll be possible to reuse the code?

**Comment by carolhmj** [comment]
@sebavan @RaananW code changed to use a single function!

File: packages/dev/core/src/Meshes/abstractMesh.ts [hunk 1]
```diff
@@ -1456,14 +1456,14 @@ export class AbstractMesh extends TransformNode implements IDisposable, ICullabl
...
-     * @returns the position data
+     * Internal function to get buffer data and possibly apply morphs and normals
+     * @param applySkeleton
+     * @param applyMorph
+     * @param data
+     * @param isNormalData
      */
-    public getPositionData(applySkeleton: boolean = false, applyMorph: boolean = false, data?: Nullable<FloatArray>): Nullable<FloatArray> {
-        data = data ?? Tools.Slice(this.getVerticesData(VertexBuffer.PositionKind));
+    private _getData(applySkeleton: boolean = false, applyMorph: boolean = false, data?: Nullable<FloatArray>, isNormalData: boolean = false): Nullable<FloatArray> {
```
Comment (RaananW) [inline packages/dev/core/src/Meshes/abstractMesh.ts:1465]: You can accept "kind" and pass the right kind to the function (normal or position)

**Review by RaananW — COMMENTED** [review]
Just me being nitpicky 😏

I'll let you decide if it makes sense, add i see you are using this flag in the function as well (not only to determine the kind)

File: packages/dev/core/src/Meshes/abstractMesh.ts (hunk 1 above)
Comment (carolhmj) [reply packages/dev/core/src/Meshes/abstractMesh.ts]: Thanks, it does look cleaner this way :D

**Review by RaananW — APPROVED** [review]

**Pull request merged**